docs/design/simple-rolling-update.md

```diff
@@ -0,0 +1,37 @@
+## Simple rolling update
+This is a lightweight design document for simple rolling update in ```kubectl```
+
+### Lightweight rollout
+Assume that we have a current replication controller named ```foo``` and it is running image ```image:v1```
+
+```kubectl rollingupdate rc foo --image=myimage:v2```
+
+Under the hood, this will do the following:
+   * Create a new replication controller named foo-next, identical to foo, except image is updated to the new image.
+   * Do the standard down/up one-by-one until foo has replicas zero and foo-next has replicas N
+   * delete rc foo
+   * create rc foo that is identical to foo-next
+   * delete foo-next
+
+Obviously there is a race here, where if you kill the client between delete foo, and creating the new version of 'foo' you might be surprised about what is there, but I think that's ok.
```

Comment:
> On Apr 10, 2015, at 11:23 PM, Brendan Burns <notifications@github.com> wrote:
> 
> In docs/design/simple-rolling-update.md:
> 
> > +## Simple rolling update
> > +This is a lightweight design document for simple rolling update in ```kubectl```
> > +
> > +### Lightweight rollout
> > +Assume that we have a current replication controller named ```foo``` and it is running image ```image:v1```
> > +
> > +```kubectl rollingupdate rc foo --image=myimage:v2```
> > +
> > +Under the hood, this will do the following:
> > +   * Create a new replication controller named foo-next, identical to foo, except image is updated to the new image.
> > +   * Do the standard down/up one-by-one until foo has replicas zero and foo-next has replicas N
> > +   * delete rc foo
> > +   * create rc foo that is identical to foo-next
> > +   * delete foo-next
> > +
> > +Obviously there is a race here, where if you kill the client between delete foo, and creating the new version of 'foo' you might be surprised about what is there, but I think that's ok.
> Really? I think it's pretty unlikely. How many people are going to control-c a rolling update that is basically 100% done?
> 
All the time, and when they do, it's because of this.

    kubectl rolling-update old new
    ... Waiting ...
    /horror oh no oh no oh no
    ctrl+c
    
So the ctrl+c because of a mistake is *exactly* at the worst time, and it is when the user will *most* need to recover.  That's when good tools are separated from great tools.  Protecting the user in this moment is the most critical moment.
> —
> Reply to this email directly or view it on GitHub.
>